cmd/podman/compose.go

```diff
@@ -85,17 +86,23 @@ func composeProvider() (string, error) {
 		return "", errors.New("no compose provider specified, please refer to `man podman-compose` for details")
 	}
 
+	var lookupErrors []error
 	for _, candidate := range candidates {
 		path, err := exec.LookPath(os.ExpandEnv(candidate))
-		if err != nil {
-			if errors.Is(err, os.ErrNotExist) {
-				continue
-			}
-			return "", err
+		if err == nil {
+			// First specified provider "candidate" wins.
+			logrus.Debugf("Found compose provider %q", path)
+			return path, nil
+		}
+		// #20502: report fatal errors in case no working provider is found
+		if !errors.Is(err, os.ErrNotExist) {
```

Comment:
I don't understand this here? The reported error in the issue is `exec.ErrNotFound`, why are we not ignoring that as well?

Right now that always leaves the user with an error `not found in $PATH` which means they think they need to put it in $PATH which is not necessarily required as any other configured path would be fine as well.
IMO it would make much more sense to return all the tested paths in the error message instead of referring users to the docs.

Reply:
> The reported error in the issue is exec.ErrNotFound, why are we not ignoring that as well?

Good catch. Yes, this error should be ignored as well.

> IMO it would make much more sense to return all the tested paths in the error message instead of referring users to the docs.

So you suggest to extend the error message(s?) below with the tested tests (i.e., `candidates`)?

Reply:
yes I think it makes more sense to list all checked paths directly in the error message instead of asking users to refer to the docs? It is not even clear where a user should look to find the paths, if one opens the podman-compose man page they find the reference to containers.conf(5) but in there the list of paths is not even documented in https://github.com/containers/common/blob/main/docs/containers.conf.5.md. So technically this is impossible to find out by only looking at the docs.

Reply:
@Luap99, I played a bit and liked the following output most:
```
podman (fix-20502) $ ./bin/podman compose up
Error: looking up compose provider failed
7 errors occurred:
        * exec: "docker-compose": executable file not found in $PATH
        * exec: "/home/vrothberg/.docker/cli-plugins/docker-compose": stat /home/vrothberg/.docker/cli-plugins/docker-compose: no such file or directory
        * exec: "/usr/local/lib/docker/cli-plugins/docker-compose": stat /usr/local/lib/docker/cli-plugins/docker-compose: no such file or directory
        * exec: "/usr/local/libexec/docker/cli-plugins/docker-compose": stat /usr/local/libexec/docker/cli-plugins/docker-compose: no such file or directory
        * exec: "/usr/lib/docker/cli-plugins/docker-compose": stat /usr/lib/docker/cli-plugins/docker-compose: no such file or directory
        * exec: "/usr/libexec/docker/cli-plugins/docker-compose": stat /usr/libexec/docker/cli-plugins/docker-compose: no such file or directory
        * exec: "podman-compose": executable file not found in $PATH
```

That'll just collect _all_ errors without any special treatment.  WDYT?

Reply:
That looks great to me. Only nit would be that the path is shown twice for each path but this seems to be caused by the error returned from LookPath so there is little would could to to change it.

Reply:
Excellent.  Thanks for the helpful review!